### DISCUSSION TITLE
A question in SmearedCracking

### DISCUSSION THREAD

**Discussion by xuxiaobei1995** [idaholab/moose#19445] [Q&A Modules: Solid mechanics] [answered]
Hi all,

     Recently I'm doing some work about cracking based on SmearedCracking. I want to realize two things:
     (1) Adjust the Young's modulus and Poisson's ratio depending on the number of cracks (For example, E_crack = E_intatct * (0.6)^N, N means the number of cracks).
     (2) When the cracked region is under compressive stresses and the temperature is above a given value, the cracks "heal" so that everything turn back to the original status.
     Because I knows little about the theory of the tensor mechanics, The code is difficult for me to understand. Does anyone know how to realize these two things? Any suggestion is appreciated.

Best Regards,
Xiaobei

**Comment by GiudGiud** [comment, marked as answer]
Hello

There s pretty good docs around smeared cracking models
https://mooseframework.inl.gov/source/materials/ComputeSmearedCrackingStress.html#!
and softening models dependent on the number of cracks
https://mooseframework.inl.gov/source/materials/PowerLawSoftening.html

Guillaume
